internal/cluster/validations/validations.go

```diff
@@ -27,6 +27,7 @@ const (
 	clusterNameRegex  = "^([a-z]([-a-z0-9]*[a-z0-9])?)*$"
 	dnsNameRegex      = "^([a-z0-9]+(-[a-z0-9]+)*[.])+[a-z]{2,}$"
 	CloudOpenShiftCom = "cloud.openshift.com"
+	sshPublicKeyRegex = "^(ssh-rsa|ssh-ed25519|ecdsa-[-a-z0-9]*) AAAA[0-9A-Za-z+/]+[=]{0,3}( [^@]+@[^@| |\\t|\\n]+)?$"
```

Comment:
doesn't this one looks better?

`#^(ssh-rsa AAAAB3NzaC1yc2|ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNT|ecdsa-sha2-nistp384 AAAAE2VjZHNhLXNoYTItbmlzdHAzODQAAAAIbmlzdHAzOD|ecdsa-sha2-nistp521 AAAAE2VjZHNhLXNoYTItbmlzdHA1MjEAAAAIbmlzdHA1Mj|ssh-ed25519 AAAAC3NzaC1lZDI1NTE5|ssh-dss AAAAB3NzaC1kc3)[0-9A-Za-z+/]+[=]{0,3}( .*)?$#`

Reply:
took from here https://gist.github.com/paranoiq/1932126